packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -51,10 +54,36 @@ class TransferEngine implements ITransferEngine {
     }
 
     throw new Error(
-      `Strapi versions doesn't match (${strategy} check): ${sourceVersion} does not match with ${destinationVersion} `
+      `Strapi versions doesn't match (${strategy} check): ${sourceVersion} does not match with ${destinationVersion}`
     );
   }
 
+  #assertSchemasMatching(sourceSchemas: any, destinationSchemas: any) {
+    const chosenStrategy = this.options.schemasMatching || 'strict';
+    const keys = uniq(Object.keys(sourceSchemas).concat(Object.keys(destinationSchemas)));
+    const diffs: { [key: string]: Diff[] } = {};
+
+    keys.forEach((key) => {
+      const sourceSchema = sourceSchemas[key];
+      const destinationSchema = destinationSchemas[key];
+      const schemaDiffs = calculateSchemaDiffs(sourceSchema, destinationSchema);
```

Comment:
It might just be a personal preference here (feel free to ignore it), but I would rather use `computeSchemaDiffs` (instead of `calculate`).

My reasoning is:
- using `calculate` for arithmetic processes
- using `compute` for algorithmic processes

Reply:
Also, with calculate/compute schema diffs, I would expect the result to be the diffs, not an object containing multiple strategies linked to the data transfer process.

Here I would maybe see more smth like:

(just a quick thought, feel free to disagree)

``` typescript
const diffs = jsonDiffs(source, destination);
const isValid = validateSchemaDiffs(diffs, strategy);
```

or

```typescript
const isValid = createSchemaDiffsValidator()
    .diff(source, destination)
    .use(strategy)
```

or

```typescript
// here strategies needs to return failing diffs instead of just true/false
const diffs = compareSchemas(strategy, source, destination);

if (diffs.length > 0) {
  // error + you have access to the diff causing the error
}
```

Reply:
I agree, my preference is for something like the third option.